### ISSUE TITLE
Error in test in test_ud_fit.R

### ISSUE THREAD

**Issue by pcarbo** [stephenslab/udr#23]
@yunqiyang0215 I'm getting an error in one of the tests:

```
Error (test_ud_fit.R:202:3): U[[i]] does not get updated when sum(P[,i]) is near zero
<subscriptOutOfBoundsError/error/condition>
Error in `U[, , j]`: subscript out of bounds
Backtrace:
    ▆
 1. ├─utils::capture.output(fit1 <- ud_fit(fit0, control = list(weights.update = "none"))) at test_ud_fit.R:202:3
 2. │ └─base::withVisible(...elt(i))
 3. └─udr::ud_fit(fit0, control = list(weights.update = "none"))
 4.   └─udr:::ud_fit_em(fit, covupdates, control, verbose) at udr/R/ud_fit.R:303:3
 5.     └─udr::compute_posterior_probs(fit, control$version) at udr/R/ud_fit.R:333:5
 6.       └─udr:::compute_posterior_probs_iid(fit$X, fit$w, U, V) at udr/R/posterior.R:112:6
 7.         └─mvtnorm::dmvnorm(X, sigma = V + U[, , j], log = TRUE) at udr/R/posterior.R:148:3
 8.           └─base::ncol(sigma)
```

Could you please take a look?

**Comment by yunqiyang0215** [comment]
@pcarbo It is due to error in the input created by this line fit0$w <- with(fit0,w/sum(w)). There are 11 weights but 10 Us. 

Also I think the way to test this "U[[i]] does not get updated when sum(P[,i]) is near zero" is not by setting weights.update = "none". I think we don't want to use weights.update = "none" unless k=1?

**Comment by pcarbo** [comment]
I see, yes, the issue is with the test. It should be fixed now; see commit id 9901337.
